src/cache.rs

```diff
@@ -0,0 +1,89 @@
+use std::collections::HashMap;
+use std::fmt;
+use std::fs::OpenOptions;
+use std::io::Read;
+use std::io::{self, Write};
+use std::sync::RwLock;
+
+pub struct SimpleStringCache {
+    pub cache: RwLock<HashMap<String, String>>,
+    pub file_path: String,
+}
+
+impl fmt::Debug for SimpleStringCache {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let cache = self.cache.read().unwrap();
+        f.debug_struct("SimpleCache")
+            .field("file_path", &self.file_path)
+            .field("cache", &*cache)
+            .finish()
+    }
+}
+
+impl SimpleStringCache {
+    pub fn new(file_path: String) -> SimpleStringCache {
+        let cache = HashMap::new();
+
+        let cache = SimpleStringCache {
+            cache: RwLock::new(cache),
+            file_path,
+        };
+
+        cache.load_from_file().unwrap();
+        cache
+    }
+
+    pub fn put(&self, key: String, value: String) -> io::Result<()> {
+        {
+            let mut cache = self.cache.write().unwrap();
+            cache.insert(key.clone(), value);
+        }
+        Ok(())
+    }
+
+    pub fn get(&self, key: &str) -> Option<String> {
+        let cache = self.cache.read().unwrap();
+        cache.get(key).cloned()
+    }
+
+    fn load_from_file(&self) -> Result<(), io::Error> {
+        let mut file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(true)
+            .open(&self.file_path)?;
+
+        let mut contents = String::new();
+        file.read_to_string(&mut contents)?;
+
+        let mut cache = self.cache.write().unwrap();
+        cache.clear();
+        for line in contents.lines() {
+            let parts: Vec<&str> = line.split(',').collect();
+            if parts.len() == 2 {
+                cache.insert(parts[0].to_string(), parts[1].to_string());
+            } else {
+                tracing::error!("Invalid format line in cache file: {}", line);
```

Comment:
Generally, it's a good idea to log information, but in this application, I haven't decided on a logging strategy yet. Also, this process clearly indicates data inconsistency, so `panic` might be sufficient for now.

Reply:
Makes sense!